screens/home.js

```diff
@@ -1,208 +1,222 @@
+import { GiftedChat } from "react-native-gifted-chat";
 
 const user = {
   _id: 1,
-  name: 'Developer',
-}
+  name: "Developer"
+};
 
 const otherUser = {
   _id: 2,
-  name: 'React Native',
-  avatar: 'https://facebook.github.io/react/img/logo_og.png',
-}
+  name: "React Native",
+  avatar: "https://facebook.github.io/react/img/logo_og.png"
+};
 
 const styles = StyleSheet.create({
-  container: { flex: 1 },
-})
-
+  container: { flex: 1 }
+});
 
 export default class Home extends Component {
   state = {
     messages: [
       {
         _id: 1,
-        text: 'This is Sleepwell. Would you like to learn more about how i can help?',
+        text:
+          "This is Sleepwell. Would you like to learn more about how i can help?",
         createdAt: new Date(),
         quickReplies: {
-            type: 'radio', // or 'checkbox',
+          type: "radio", // or 'checkbox',
+          keepIt: true,
+          values: [
+            {
+              title: "😋 Yes",
+              value: "yes"
+            },
+            {
+              title: "📷 Yes,show me with a picture!",
+              value: "yes_picture"
+            },
+            {
+              title: "😞 Nope. What?",
+              value: "no"
+            }
+          ]
+        },
+        user: {
+          _id: 2,
+          name: "React Native"
+        }
+      }
+    ],
+    typingText: null,
+  };
+
+  onReceive = text => {
+    this.setState(previousState => {
+      return {
+        messages: GiftedChat.append(
+          previousState.messages,
+          {
+            _id: Math.round(Math.random() * 1000000),
+            text,
+            createdAt: new Date(),
+            user: otherUser
+          },
+          Platform.OS !== "web"
+        ),
+      };
+    });
+  };
+
+  onSend = (messages = []) => {
+    const step = this.state.step + 1;
+    this.setState(previousState => {
+      const sentMessages = [{ ...messages[0], sent: true, received: true }];
+      return {
+        messages: GiftedChat.append(
+          previousState.messages,
+          sentMessages,
+          Platform.OS !== "web"
+        ),
+        typingText: "Chat bot is typing...",
+        step
+      };
+    });
+    // for demo purpose
+    // setTimeout(() => this.botSend(step, messages[0]), Math.round(Math.random() * 1000))
+  };
+  
+  onSendFromUser = (messages = []) => {
+    const createdAt = new Date();
+    const messagesToUpload = messages.map(message => ({
+      ...message,
+      user,
+      createdAt,
+      _id: Math.round(Math.random() * 1000000)
+    }));
+    this.onSend(messagesToUpload);
+  };
+
+  onQuickReply = replies => {
+    const createdAt = new Date();
+    if (replies.length === 1) {
+      this.onSend([
+        {
+          createdAt,
+          _id: Math.round(Math.random() * 1000000),
+          text: replies[0].title,
+          user
+        }
+      ]);
+      setTimeout(() => this.determineResponse(replies[0]), 2000);
```

Comment:
Although I think 2 seconds is a bit long as a delay will reduce user experience. Maybe we can decrease it in the future? I think 1 second is a good enough wait time. Up to you

Reply:
i agree, lets play around with this value in the future